### ISSUE TITLE
[generic type] CtMethod.getReference().getActualMethod() fails for generic parameter types

### ISSUE THREAD

**Issue by pvojtechovsky** [INRIA/spoon#1256]
```java
//model
interface IFace<T extends Long> {
 void method(T param);
}
//test
CtMethod method = ...//refers `void method(T param)`
method.getReference().getActualMethod(); //returns null
```
method.getReference().getActualMethod() returns null, because parameter type reference `T` is not able to found it's declaration in `IFace`, because CtExecutableReference parent is null - there is no way how to get to declaration of generic type dynamically.

The workaround is:
```java
CtExecutableReference ref = method.getReference();
ref.setParent(method);
ref.getActualMethod(); //returns expected `Method`
```
I have seen this problem [here](https://github.com/INRIA/spoon/pull/1218/commits/ffab37263d416be0476cd26ebb23c07a6724deb4#diff-89d21fcd8875342a0be9725d0bb210f9R54)

**Comment by monperrus** [comment]
should we encode the workaround in `getReference` dierctly?

**Comment by pvojtechovsky** [comment]
The workaround `ref.setParent(method)` was already added [here](https://github.com/INRIA/spoon/blob/27fc8208c4da4145c687c7f4db057de7dbed50f6/src/main/java/spoon/reflect/factory/ExecutableFactory.java#L101). But is just workaround, because we cannot assure that parent of executable reference will always point to it's declaring method. I have created a test and "some" fix in #1277

**Comment by pvojtechovsky** [comment]
The problem is still there. It can be reproduce on this model
```java
//model
interface IFace<T extends Long> {
 <U extends InputStream> void method(T param, U param2);
}
//test
CtMethod method = ...//refers `void method(T param)`
method.getReference().getActualMethod(); //returns null
```
It is reproduced on PR #1286. Please try to solve this problem, because I did not found good solution. I think that we should change spoon model to fix it correctly. But may be you will found another workaround.
I do not need solution urgently. I just wanted to point at this spoon concept problem of automatic computation of scope of generic type parameter, which is bad idea from my point of view.

Note: JDT compiler provides scope for each generic type parameter, but spoon ignores it - hoping that it is always possible to detect the scope from the context.

Please reopen this issue. I do not know how to reopen, or have no permission for that.

**Comment by monperrus** [comment]
closed per #1286
